core/state/statedb_arbitrum.go

```diff
@@ -233,16 +240,13 @@ func (s *StateDB) StartRecording() {
 	s.arbExtraData.userWasms = make(UserWasms)
 }
 
-func (s *StateDB) RecordProgram(moduleHash common.Hash) {
-	asm, err := s.TryGetActivatedAsm(moduleHash)
+func (s *StateDB) RecordProgram(targetNames []string, moduleHash common.Hash) {
```

Comment:
add here a small condition that if len(targetNames) ==0 -> do nothing and return.
Eventually, that's what we'll use to avoid unnecessary work when recording.